### DISCUSSION TITLE
쿠폰 할인 클래스 스펙

### DISCUSSION THREAD

**Discussion by Bue-von-hon** [f-lab-edu/foody-lab#9] [Show and tell]
안녕하세요 저는 쿠폰 할인 기능 개발을 진행하고 있습니다. #4 
다름이 아니라 해당 클래스가 갖고있는 역할과 테스트 범위에 대한 저의 생각을 간략하게 기록하려고합니다.

쿠폰 할인 클래스는 아래와 같은 속성을 갖고 있습니다.
1. 할인할 금액
2. 쿠폰 유효 기간

테스트 범위는 아래와 같습니다.
1. 유효한 할인 금액인지 테스트(할인 금액은 이 클래스에서 잡아야 편할것 같습니다)
2. 쿠폰 유효기간

아쉬운 점은 퍼센티지 할인이 아니라 금액할인이라는 점입니다.
이 부분은 추후에 추가하면 좋을 것 같습니다.

**Comment by Bue-von-hon** [comment]
@Taewoongjung @f-lab-Carrey 쿠폰 할인 클래스에 대해 편하게 의견부탁드립니다!

**Reply by Bue-von-hon** [reply]
할인 정책도 퍼센트 할인/ 금액 할인 이런식으로 있을 수 있어보여요
현재는 금액 할인만 구현되어 있는 부분도 인터페이스로 바꿔볼 수 있어보입니다!

**Reply by Taewoongjung** [reply]
네 인터페이스로 빼야한다고 봅니다. 
그래야 OCP 위반을 안하고 변경에 유연할거라고 봐요.

**Comment by f-lab-Carrey** [comment]
github에 이런 기능이 있는지 몰랐네요~ 두분께서 건설적인 의견을 주고 받으시는 것 같아 좋습니다~!😁
